### DISCUSSION TITLE
Error location (v0.4.1)

### DISCUSSION THREAD

**Discussion by Beliar83** [biojppm/rapidyaml#232] [Q&A] [answered]
I have the following error handling function (ERR_PRINT is from godot):

```
	std::stringstream errorString;
	auto messageString = std::string(msg, len);
	auto nameString = std::string(loc.name.str, loc.name.len);
	errorString << "Error parsing yaml: " << nameString << "(" << loc.line + 1 << ", " << loc.col + 1 << ") (" << loc.offset << "): " << messageString;
	ERR_PRINT(errorString.str().c_str());
```

The output I get from that is this:
`ERROR: Error parsing yaml: E:\projekte\godot-yaml\rapidyaml-install\include\c4/yml/tree.hpp(578, 1) (0): check failed: has_val(node)`

I would have thought that I get the location inside the yaml file/string, not the source/header. This is not really helpful. Is this how it is supposed to be?

Edit:
Although, maybe the problem is that this is not inside the yaml file itself, but from my code... though knowing where exactly would be helpful.

**Comment by Beliar83** [comment]
I think I should rather find out how I can get debugging set up for gdnative

**Reply by biojppm** [reply]
[PASTED AS ANSWER BELOW]

It depends on the phase:
  * When an error occurs while ryml is parsing the YAML code (ie, a legit YAML syntax error), the callback does receive the location on the YAML file (if you use the proper `ryml::parse()` overload).
  * Thereafter, when the error occurs from code you wrote to go through the data (ie, YAML is syntactically correct but the data is semantically wrong, or the c++ code does not match the data), then of course the failure cannot be implicitly traced to the YAML source. So you receive the C++ location where the error was detected.
  * But you can still get the YAML location using the locations feature to find where in the source a given node was. But there are some caveats. [See the quickstart](https://github.com/biojppm/rapidyaml/blob/213b201d264139cd1b887790197e08850af628e3/samples/quickstart.cpp#L3798) to discover how it works and how it should be used.

**Reply by Beliar83** [reply]
Ah, thanks I will take a look at that.

**Comment by biojppm** [comment, marked as answer]
It depends on the phase:
  * When an error occurs while ryml is parsing the YAML code (ie, a legit YAML syntax error), the callback does receive the location on the YAML file (if you use the proper `ryml::parse()` overload).
  * Thereafter, when the error occurs from code you wrote to go through the data (ie, YAML is syntactically correct but the data is semantically wrong, or the c++ code does not match the data), then of course the failure cannot be implicitly traced to the YAML source. So you receive the C++ location where the error was detected.
  * But you can still get the YAML location using the locations feature to find where in the source a given node was. But there are some caveats. [See the quickstart](https://github.com/biojppm/rapidyaml/blob/213b201d264139cd1b887790197e08850af628e3/samples/quickstart.cpp#L3798) to discover how it works and how it should be used.
